src/fragments/lib/push-notifications/react-native/receive_device_token/on-token-received.mdx

```diff
@@ -1,9 +1,36 @@
+<BlockSwitcher>
+<Block name="TypeScript">
+
+```ts
+import {
+  onTokenReceived,
+  OnTokenReceivedInput,
+  OnTokenReceivedOutput
+} from 'aws-amplify/push-notifications';
+
+const myTokenReceivedHandler: OnTokenReceivedInput = (token) => {
+  // Do something with the received token
+};
+
+const listener: OnTokenReceivedOutput = onTokenReceived(myTokenReceivedHandler);
+
+listener.remove(); // Remember to remove the listener when it is no longer needed
+```
+
+</Block>
+<Block name="JavaScript">
+
 ```js
+import { onTokenReceived } from 'aws-amplify/push-notifications';
+
 const myTokenReceivedHandler = (token) => {
   // Do something with the received token
 };
 
-const listener = Notifications.Push.onTokenReceived(myTokenReceivedHandler);
+const listener = onTokenReceived(myTokenReceivedHandler);
```

Comment:
Maybe it's elsewhere and I am just not seeing it here but do we want to mention when this is invoked? Like in which app lifecycle states? Remember this being a confusing piece of information to people in the previous version of the category.

Reply:
It is in a fragment that already exists in `/lib`

Reply:
Got it, thanks!